openapi-cli/src/main/java/io/ballerina/generators/BallerinaSchemaGenerator.java

```diff
@@ -258,13 +257,33 @@ private TypeDefinitionNode getTypeDefinitionNode(OpenAPI openAPI,
                     typeDefinitionNodeList.add(typeDefNode);
                 }
             }
+        } else if (schemaValue.get$ref() != null) {
+            Token typeRef = AbstractNodeFactory.createIdentifierToken(getValidName(
+                    extractReferenceType(schemaValue.get$ref()), true));
+            Token asterisk = AbstractNodeFactory.createIdentifierToken("*");
+            Token semicolon = AbstractNodeFactory.createIdentifierToken(";");
+            TypeReferenceNode recordField =
+                    NodeFactory.createTypeReferenceNode(asterisk, typeRef, semicolon);
+            recordFieldList.add(recordField);
+            NodeList<Node> fieldNodes = AbstractNodeFactory.createNodeList(recordFieldList);
+            RecordTypeDescriptorNode recordTypeDescriptorNode =
+                    NodeFactory.createRecordTypeDescriptorNode(createToken(SyntaxKind.RECORD_KEYWORD),
+                            createToken(OPEN_BRACE_TOKEN), fieldNodes, null,
+                            createToken(SyntaxKind.CLOSE_BRACE_TOKEN));
+            MarkdownDocumentationNode documentationNode =
+                    createMarkdownDocumentationNode(createNodeList(schemaDoc));
+            MetadataNode metadataNode = createMetadataNode(documentationNode, createEmptyNodeList());
+            typeDefNode = NodeFactory.createTypeDefinitionNode(metadataNode,
+                    null, typeKeyWord, typeName, recordTypeDescriptorNode, createToken(SEMICOLON_TOKEN));
+            typeDefinitionNodeList.add(typeDefNode);
+
```

Comment:
Shall we add a small test scenario to handle  nested reference in record  ?

Reply:
Added